pkg/operator/watchdog.go

```diff
@@ -70,11 +87,11 @@ func (dw *DaprWatchdog) Start(parentCtx context.Context) error {
 				if !ok {
 					continue
 				}
-				ok = dw.listPods(ctx)
-				if firstCompleteCh != nil {
+				ok = dw.listPods(ctx, sel)
+				if !firstCompleted {
```

Comment:
Why this change? Checking for channel being nil should have been enough

Reply:
there was a race condition